p-token/src/processor/shared/initialize_mint.rs

```diff
@@ -71,50 +71,39 @@ pub fn process_initialize_mint(
 }
 
 /// Instruction data for the `InitializeMint` instruction.
-pub struct InitializeMint<'a> {
-    raw: *const u8,
+#[repr(C)]
+struct InitializeMint {
+    pub(crate) decimals: u8,
 
-    _data: PhantomData<&'a [u8]>,
+    pub(crate) mint_authority: Pubkey,
+
+    freeze_authority: (u8, Pubkey),
 }
 
-impl InitializeMint<'_> {
+impl InitializeMint {
     #[inline]
-    pub fn try_from_bytes(bytes: &[u8]) -> Result<InitializeMint, ProgramError> {
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<&InitializeMint, ProgramError> {
         // The minimum expected size of the instruction data is either 34 or 66 bytes:
         //   - decimals (1 byte)
         //   - mint_authority (32 bytes)
         //   - option + freeze_authority (1 byte + 32 bytes)
-        if bytes.len() < 34 || (bytes[33] == 1 && bytes.len() < 66) {
-            return Err(ProgramError::InvalidInstructionData);
+        unsafe {
+            match bytes.len() {
+                34 if *bytes.get_unchecked(33) == 0 => {
+                    Ok(&*(bytes.as_ptr() as *const InitializeMint))
+                }
+                66 => Ok(&*(bytes.as_ptr() as *const InitializeMint)),
```

Comment:
Should this have `if *bytes.get_unchecked(33) == 1` like the other one?

Reply:
Same here, we could have the check on the accessor instead.